security/WFCORE-5466_Remove_Legacy_Vault_Integration.adoc

```diff
@@ -0,0 +1,311 @@
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+At the start for JBoss AS7 the secure handling of credentials in the management model was handled
+by the PicketBox vault allowing expressions in the management model to be resolved against the 
+vault.
+
+During WildFly 11 with the addition of support for WildFly Elytron an alternative solution was
+added providing a credential store which can be used to resolve credentials.  This was later
+enhanced to also support the use of encrypted expressions in the management model in general.
+
+This task is focused on the removal of the PicketBox based vault integration.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5466[WFCORE-5466 - Remove Vault support entirely from WildFly Core]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFCORE-5485[WFCORE-5485 - Bump the CLI configuration schema to version 4]
+* https://issues.redhat.com/browse/WFLY-14912[WFLY-14912 - Remove Vault support entirely from WildFly]
+* https://issues.redhat.com/browse/EAP7-1745[EAP7-1745 - Remove Vault support]
+* https://issues.redhat.com/browse/WFLY-14943[WFLY-14943 - Once all legacy security removal is in perform a full refactor]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE this does not require new feature test development, however both teams will need
+to review existing testsuites for tests dependent on the Vault integration.  Identified tests will
+either need deleting or converting to a credential store equivalent.
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [ ] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+As a removal task, this section is describing the steps being undertaken rather than a set of
+"requirements".
+
+=== Hard Requirements
+
+The command line vault utility scripts `vault.bat`, `vault.ps1`, and `vault.sh` are all removed
+from the server.
+
+The next version of the management model is updated to remove all vault resources.  Any use of
+vault resources in the latest version will result in a usual unexpected element error.
+
+Where configurations are read using older model versions if they contain a vault definition this
+will be logged as a warning and the vault configuration dropped at the next write.  This is to
+allow older configurations to still be handled in admin mode.  This same approach will be used for
+appclient parsing.
+
+Vault expressions are not prevented from occurring within the management model for two reasons:
```

Comment:
Do we plan to provide an automated way to migrate vault expression occurrences into credential store based values and encrypted expressions?